cns/fsnotify/fsnotify.go

```diff
@@ -114,7 +117,25 @@ func (w *watcher) watchFS(ctx context.Context) error {
 	err = watcher.Add(w.path)
```

Comment:
maybe add a comment here that it's critical the watcher starts here before we do the one time read otherwise it could be possible to miss one if the watcher is started after the read.